engines/bastion_katello/app/assets/javascripts/bastion_katello/content-hosts/bulk/views/content-hosts-bulk-system-purpose-modal.html

```diff
@@ -0,0 +1,80 @@
+<div data-extend-template="components/views/bst-modal.html">
+  <h4 data-block="modal-header" translate>Content Host System Purpose</h4>
+
+  <div data-block="modal-body">
+    <h4 translate>Assign System Purpose:</h4>
+
+    <div class="row">
+      <div class="col-sm-12">
+        <div bst-global-notification></div>
+      </div>
+    </div>
+
+    <div bst-alert="info" ng-show="showConfirm">
+      <span translate>
+        Are you sure you want to set the System Purpose values on the content host(s) selected?
+      </span>
+      <div>
+        <button type="button" class="btn btn-default" ng-click="showConfirm = false; performAction()" translate>Yes</button>
+        <button type="button" class="btn btn-default" ng-click="showConfirm = false; cancel()" translate>No</button>
+      </div>
+    </div>
+
+    <form name="contentHostContentForm" class="form" ng-hide="content.workingMode">
+      <div>
+        <label translate>Service Level:</label>
+
+        <select type="select"
+                ng-options="service_level for service_level in defaultServiceLevels"
+                ng-model="selected.service_level">
+                <option value="" ng-if="false"></option>
+        </select>
+      <br /><br />
+
+        <label translate>Usage Type:</label>
+
+        <select type="select"
+                readonly="denied('edit_hosts', host)"
+                ng-options="usage_type for usage_type in defaultUsages"
+                ng-model="selected.purpose_usage">
+                <option value="" ng-if="false"></option>
+        </select>
+        <br /><br />
+
+        <label translate>Role:</label>
+
+        <select type="select"
+                readonly="denied('edit_hosts', host)"
+                ng-options="roles for roles in defaultRoles"
+                ng-model="selected.purpose_role">
+                <option value="" ng-if="false"></option>
+        </select>
+        <br /><br />
+
+        <label translate>Add ons:</label>
```

Comment:
Need to make sure that leaving this blank won't clear all add-ons.

Reply:
(to be clear, I didn't test this)

Reply:
I fixed this by default setting the value to `null`
https://github.com/Katello/katello/pull/8989/files#diff-36d24e3abb899cdb34ffec17f2a4458ce1baa7efeb5540fc5ba40abf3a01e87cR70

Which does not take any action from the unit test here:

https://github.com/Katello/katello/pull/8980/files#diff-19db80ef1ed1c8bebb89697baa0db9bc664be7f730d1301d874b99af864e813dR32